Source/CompSpinalEngineTrail.cs

```diff
@@ -0,0 +1,179 @@
+                {
+                    return (int)(parent.TryGetComp<CompPowerTrader>().Props.PowerConsumption);
+                }
+            }
+        }
+        public override int FuelUse
+        {
+            get
+            {
+                if (parent.TryGetComp<CompSpinalEngineTrail>() == null) //should never? happen but reflex
+                {
+                    Log.Message("SOS2 spinal engines: somehow hit FuelUse from a non-compspinalenginetrail?");
+                    return base.FuelUse;
+                }
+                if (fullyFormed)
+                {
+                    //Log.Message($"sos2 spinal engines thrust called about a spinal engine, returning cached fuelUse {cachedFuelUse}");
+                    return cachedFuelUse;
+                }
+                else
+                {
+                    //Log.Message($"sos2 spinal engines fuel called about a spinal engine, not fully formed");
+                    calc();
+                    return cachedFuelUse;
+                }
+            }
+        }
+        public int SupportWeight
+        {
+            get
+            {
+                if (parent.TryGetComp<CompSpinalEngineTrail>() == null) //should never? happen but reflex
+                {
+                    Log.Message("SOS2 spinal engines: somehow hit supportWeight from a non-compspinalenginetrail?!!!!??");
+                    return 0;// is this a log.error? Something went squirelly 
+                }
+                if (fullyFormed)
+                {
+                    return cachedMass;
+                }
+                else
+                {
+                    calc();
+                    return cachedMass;
+                }
+            }
+        }
+
+        public void Reset()
+        {
+            //Log.Message($"SOS2spinal engines: resetting: thrust {cachedThrust} mass {cachedMass} fueluse {cachedFuelUse} fuelallowed {cachedFuelAllowed} power {cachedPowerUse}");
+            cachedThrust = 0;
+            cachedMass = 0;
+            cachedFuelUse = 0;
+            cachedFuelAllowed = 0;
+            cachedPowerUse = 0;
+            cachedError = "Spinal engine isn't fully formed! Check for engine accelerators, and all three components of the fuel infrastructure!";
+        }
+        private void calc()
+        {
+            ReturnValue wasReturned = SOS2SpinalEngines.SpinalRecalc(this);
+            if (wasReturned == null) // null basically means something went wrong: Multiblock broken, re-set to zeros
+            {
+                Reset();
+                return;
+            }
+            else if(wasReturned.fullyFormed == false)
+            {
+                cachedError = wasReturned.playerError; //cleanest way I could think of to surface errors? 
+            }
+            else
+            {
+                Reset(); //I can't see this being needed but eh 
+                cachedThrust = (int)(base.Props.thrust * (1 + wasReturned.thrustAmp)); 
```

Comment:
hrm this should? be identical behaviour here no? MS says "When you convert a double or float value to an integral type, this value is rounded towards zero to the nearest integral value" so a positive number should just truncate the decimal portion? Which is a bit different from what I expected from cobwebby C++ memories of "it literally truncates the decimal" tbf. is there some circumstance this could round up? 🤔  Or should I consider using Floor just to be more explicit what I want, hrm

Reply:
Hmm, I'd have to test it out honestly.  It looks like c++ also truncates it so maybe I was just being panicky about how it could round thinking the machine would either way.  Though come to think of it, from a float architecture stand point, it's WAY faster to just drop the decimal part.

Reply:
I'd say it's fine as is - never mind me :p